mcsema/BC/Instruction.cpp

```diff
@@ -41,6 +42,50 @@
 
 #include "mcsema/CFG/CFG.h"
 
+namespace {
+
+// Load the address of a register.
+static llvm::Value *LoadRegAddress(llvm::BasicBlock *block,
+                                   std::string reg_name) {
+  return new llvm::LoadInst(
+      remill::FindVarInFunction(block->getParent(), reg_name), "", block);
+}
+
+// Load the value of a register.
+static llvm::Value *LoadRegValue(llvm::BasicBlock *block,
+                                 std::string reg_name) {
+  return new llvm::LoadInst(LoadRegAddress(block, reg_name), "", block);
+}
+
+static llvm::Value *LoadAddressRegValOrZero(llvm::BasicBlock *block,
```

Comment:
I think you should rename this because it's not really "or zero" if the `zero` value can be non-zero. Perhaps it can be `...OrDefault`?